x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java

```diff
@@ -52,8 +52,10 @@ public class SearchApplication implements Writeable, ToXContentObject {
     public static final String NO_TEMPLATE_STORED_WARNING = "Using default search application template which is subject to change. "
         + "We recommend storing a template to avoid breaking changes.";
 
+    public static final String NO_INDICES_WARNING = "Indices are missing. "
+        + "We recommend ensuring that the search application contains at least one index.";
     private final String name;
-    private final String[] indices;
+    private String[] indices;
```

Comment:
Maybe adding `indices` as final, but nullable, would be a better option. Providing separate constructors for providing indices or not could work as well.

The main idea is making explicit the choice of including indices or not on object creation.

WDYT?

Reply:
Updated